src/http/proxy_wasm/ngx_http_proxy_wasm_dispatch.c

```diff
@@ -240,11 +243,23 @@ ngx_http_proxy_wasm_dispatch(ngx_proxy_wasm_exec_t *pwexec,
         }
     }
 
+#if (NGX_HAVE_UNIX_DOMAIN)
+    if (host->len < uds_prefix.len ||
+        !ngx_str_eq(p, uds_prefix.len, uds_prefix.data, uds_prefix.len))
```

Comment:
Same here, this feels like it could be a single `ngx_str_eq`, without the length comparison above. Reminder that `ngx_str_eq` already checks for lengths matching before doing any comparison of bytes.